base/linalg/eigen.jl

```diff
@@ -90,11 +90,17 @@ end
 #Computes maximum and minimum eigenvalue
 function eigmax(A::Union{Number, StridedMatrix}; permute::Bool=true, scale::Bool=true)
     v = eigvals(A, permute = permute, scale = scale)
-    iseltype(v,Complex) ? error("DomainError: complex eigenvalues cannot be ordered") : maximum(v)
+    if iseltype(v,Complex)
+      throw(DomainError())
```

Comment:
I think it can make an exception much more costly if contains a string (or any other pointer to a Julia object). These costs are probably okay for expensive linear algebra functions but not in scalar math functions such as `sqrt` where `DomainError` is used.